client/lobby/SelectionTab.cpp

```diff
@@ -115,6 +115,27 @@ SelectionTab::SelectionTab(ESelectionScreen Type)
 	: CIntObject(LCLICK | WHEEL | KEYBOARD | DOUBLECLICK), callOnSelect(nullptr), tabType(Type), selectionPos(0), sortModeAscending(true)
 {
 	OBJ_CONSTRUCTION;
+
+	// initialize 'generalSortingBy' early
+	switch(tabType)
+	{
+	case ESelectionScreen::newGame:
+		generalSortingBy = ESortBy::_name;
+		break;
+	case ESelectionScreen::loadGame:
+		generalSortingBy = ESortBy::_fileName;
+		break;
+	case ESelectionScreen::saveGame:
+		generalSortingBy = ESortBy::_fileName;
+		break;
```

Comment:
These two cases do the same thing. We can combine them. Same for other two cases. Also might be worth to extract this thing to a method. The constructor itself is already quite big.

Reply:
Sounds good. Moved out to a separate helper.